docs/authorization/authority.md

```diff
@@ -0,0 +1,127 @@
+# Authority management
+Netgrif Application Engine implements authority objects to manage access
+to resources, to protect resources from unauthorized access. Then these 
+authorities can be assigned to users to provide access to those secured
+resources.
+
+## Authorizing objects
+Authorizing objects are Java enum values, that represent authorities,
+that are needed to access a resource or use an action. E.g. to be able to import new 
+process (Petri Net) into the application, the user must have authority 
+created from `AuthorizingObject.PROCESS_UPLOAD` authorizing object.
+
+These authorizing objects are predefined, and they are used to create the 
+`Authority` objects/entities at application startup. There followings are predefined:
+
+- `PROCESS_UPLOAD`- to import new process
+- `PROCESS_VIEW_ALL`- to retrieve all the processes imported by any user
+- `PROCESS_VIEW_MY`- to retrieve only processes imported by logged user
+- `PROCESS_DELETE_ALL`- to delete processes imported by any user
+- `PROCESS_DELETE_MY`- to delete processes imported by logged user
+- `FILTER_UPLOAD`- to upload filter
+- `FILTER_DELETE_MY`- to delete filter imported by logged user
+- `FILTER_DELETE_ALL`- to delete filter imported by any user
+- `USER_CREATE`- to invite or create user
+- `USER_DELETE`- to remove user
+- `USER_EDIT_ALL`- to edit any user
+- `USER_EDIT_MY`- to edit only logged user
+- `USER_VIEW_ALL`- to retrieve all users
+- `USER_VIEW_MY`- to retrieve only logged user
+- `GROUP_CREATE`- to create group
+- `GROUP_DELETE_MY`- to delete group created by logged user
+- `GROUP_DELETE_ALL`- to delete group create by any user
+- `GROUP_ADD_USER`- to add any user to any group
```

Comment:
In demo application the user only can add users to its own group and we have defined concept of "my" groups in case GROUP_VIEW_MY so there is an endpoint to retrieve the groups owned by user. Hence I can define GROUP_ADD_USER_MY for future use.